driver/registry_default.go

```diff
@@ -815,16 +816,13 @@ func (m *RegistryDefault) PrometheusManager() *prometheus.MetricsManager {
 	return m.pmm
 }
 
-func (m *RegistryDefault) HTTPClient(ctx context.Context, opts ...httpx.ResilientOptions) *retryablehttp.Client {
+func (m *RegistryDefault) HTTPClient(_ context.Context, opts ...httpx.ResilientOptions) *retryablehttp.Client {
 	opts = append(opts,
 		httpx.ResilientClientWithLogger(m.Logger()),
 		httpx.ResilientClientWithMaxRetry(2),
-		httpx.ResilientClientWithConnectionTimeout(30*time.Second))
-
-	tracer := m.Tracer(ctx)
-	if tracer.IsLoaded() {
-		opts = append(opts, httpx.ResilientClientWithTracer(tracer.Tracer()))
-	}
+		httpx.ResilientClientWithConnectionTimeout(30*time.Second),
+		httpx.ResilientClientWithTracer(noop.NewTracerProvider().Tracer("Ory Kratos")), // will use the tracer from a context if available
```

Comment:
How can this use the tracer form the context if the context is not even passed to it?

Reply:
The tracer is retrieved from the request context:
```
req, err := http.NewRequest(ctx /* HERE */, GET, "http://localhost", nil)
```

Reply:
Ah so here we instantiate the client, and the tracer does not matter. When the request is made, a context with the correct tracer is passed.
Correct?

Reply:
Precisely. The `otelhttp` instrumentation checks for an existing span in the context. If it finds one, it uses the same tracer to start its own spans. If it doesn't find one, it uses the one passed here. Finally, it falls back to the global tracer (which we don't typicall configure).